src/bin/ls.rs

```diff
@@ -203,14 +202,136 @@ fn main() {
     }
 
 
+    let mut output = Vec::new();
     if parser.args.is_empty() {
-        list_dir(".", &parser, &mut stdout, &mut stderr);
+        list_dir(".", &parser, &mut output, &mut stderr);
     } else {
         for dir in parser.args.iter() {
-            list_dir(&dir, &parser, &mut stdout, &mut stderr);
+            list_dir(&dir, &parser, &mut output, &mut stderr);
         }
     }
+
+    if parser.found("long-format") {
+        for (i, word) in output.iter().enumerate() {
+            stdout.write(word.as_bytes()).try(&mut stderr);
+            if i % 4 == 3 {
+                !stdout.write("\n".as_bytes()).try(&mut stderr);
+            }
+        }
+    } else {
+        print_columns(output);
+    }
+    stdout.flush().try(&mut stderr);
+}
+
+/*
+fn print_columns(words: Vec<String>) {
```

Comment:
oh, i forgot to delete all the code i moved into the lib.rs file. should i fix it and make a new pull request?